control/iosys.py

```diff
@@ -925,23 +911,34 @@ def __init__(self, syslist, connections=[], inplist=[], outlist=[],
                 warn("Duplicate object found in system list: %s" % str(sys))
             elif sys.name is not None and sys.name in sysname_list:
                 warn("Duplicate name found in system list: %s" % sys.name)
+                
             sysobj_list.append(sys)
             sysname_list.append(sys.name)
 
+        # This won't work when there are duplicate sysnames
+        # Should we throw an error rather than just warning above?
+        # Users can use sys.copy() to get a unique name for the dup system.
+        if states is None:
+            states = []
+            for sys in sysobj_list:
+                states += [sys.name + '.' + statename for statename in sys.state_index.keys()]
+
```

Comment:
@murrayrm This is the part that would break if you add a named signal with a generic 'sys[i]' that will cause a conflict (it's no different from having two systems with the same name). The only "bad" thing that happens is that the system will be missing the state names of the systems with conflicting names (only the first 'sys[i].states' will be present).
Right now we are only raising a warning. As I mention in the comment, users can use sys.copy() first to create a unique name. Is this ok?